libs/agno/agno/session/team.py

```diff
@@ -132,10 +134,17 @@ def get_messages(
             skip_statuses: Skip messages with these statuses.
             skip_history_messages: Skip messages that were tagged as history in previous runs.
             skip_member_messages: Skip messages created by members of the team.
+            filter_compressed: Filter out messages that have already been compressed.
```

Comment:
Ok we just have to make sure the stored messages correctly reflect what was actually sent to the model. Since you are basically changing what is retrieved from run messages?